src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java

```diff
@@ -1,207 +1,123 @@
-            }
-        }
-        return wordEncrypted.toString();
+    private ColumnarTranspositionCipher() {
     }
 
     /**
-     * Encrypts a certain String with the Columnar Transposition Cipher Rule
+     * Encrypts a message using the Columnar Transposition Cipher with the given key.
      *
-     * @param word Word being encrypted
-     * @param keyword String with keyword being used
-     * @param abecedarium String with the abecedarium being used. null for
-     * default one
-     * @return a String with the word encrypted by the Columnar Transposition
-     * Cipher Rule
+     * @param message The message to encrypt.
+     * @param key The key to use for encryption.
+     * @return The encrypted message.
      */
-    public static String encrpyter(String word, String keyword, String abecedarium) {
-        ColumnarTranspositionCipher.keyword = keyword;
-        ColumnarTranspositionCipher.abecedarium = Objects.requireNonNullElse(abecedarium, ABECEDARIUM);
-        table = tableBuilder(word);
-        Object[][] sortedTable = sortTable(table);
-        StringBuilder wordEncrypted = new StringBuilder();
-        for (int i = 0; i < sortedTable[0].length; i++) {
-            for (int j = 1; j < sortedTable.length; j++) {
-                wordEncrypted.append(sortedTable[j][i]);
+    public static String encrypt(String message, String key) {
+        int numColumns = key.length();
+        int numRows = (int) Math.ceil((double) message.length() / numColumns);
+
+        char[][] grid = new char[numRows][numColumns];
+
+        // Fill the grid with the message and padding characters
+        fillGrid(message, grid, numRows, numColumns);
+
+        // Get the sorted column order based on the key
+        Integer[] order = getColumnOrder(key);
+
+        // Read columns in sorted order to build the encrypted message
+        StringBuilder encryptedMessage = new StringBuilder();
+        for (int col : order) {
+            for (int row = 0; row < numRows; row++) {
+                encryptedMessage.append(grid[row][col]);
             }
         }
-        return wordEncrypted.toString();
+
+        return encryptedMessage.toString();
     }
 
     /**
-     * Decrypts a certain encrypted String with the Columnar Transposition
-     * Cipher Rule
+     * Decrypts a message that was encrypted using the Columnar Transposition Cipher with the given key.
      *
-     * @return a String decrypted with the word encrypted by the Columnar
-     * Transposition Cipher Rule
+     * @param message The encrypted message.
+     * @param key The key used for encryption.
+     * @return The decrypted message.
      */
-    public static String decrypter() {
-        StringBuilder wordDecrypted = new StringBuilder();
-        for (int i = 1; i < table.length; i++) {
-            for (Object item : table[i]) {
-                wordDecrypted.append(item);
+    public static String decrypt(String message, String key) {
```

Comment:
```suggestion
    public static String decrypt(final String message, final String key) {
```

Reply:
hey, I am currently working on #5223, I think the decrypt method does not need any parameters, because we have included global variables to store the respective..